instrumentation/ktor/ktor-2-common/library/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v2_0/common/client/internal/Experimental.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.ktor.v2_0.common.client.internal
+
+import io.opentelemetry.instrumentation.ktor.v2_0.common.client.AbstractKtorClientTelemetryBuilder
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+import java.util.logging.Level
+import java.util.logging.Logger
+
+/**
+ * This class is internal and experimental. Its APIs are unstable and can change at any time. Its
+ * APIs (or a version of them) may be promoted to the public stable API in the future, but no
+ * guarantees are made.
+ */
+class Experimental {
+
+  companion object {
+    private val logger: Logger = Logger.getLogger(Experimental::class.java.name)
+
+    private val emitExperimentalTelemetryMethod = getEmitExperimentalTelemetry()
+
+    fun setEmitExperimentalTelemetry(builder: AbstractKtorClientTelemetryBuilder?, emitExperimentalTelemetry: Boolean) {
+      if (emitExperimentalTelemetryMethod != null) {
```

Comment:
Kotlin has `internal` methods that can be used only from the current module. Perhaps could skip reflection since `emitExperimentalHttpClientMetrics` is internal?